script/make-firmware.bash

```diff
@@ -29,98 +29,312 @@
 
 set -euxo pipefail
 
-echo "OUTPUT_ROOT=${OUTPUT_ROOT?}"
-
-readonly PLATFORM=nrf52840
-
-readonly BUILD_1_2_OPTIONS=('-DOT_BOOTLOADER=USB'
-  '-DOT_REFERENCE_DEVICE=ON'
-  '-DOT_BORDER_ROUTER=ON'
-  '-DOT_SERVICE=ON'
-  '-DOT_COMMISSIONER=ON'
-  '-DOT_JOINER=ON'
-  '-DOT_MAC_FILTER=ON'
-  '-DDHCP6_SERVER=ON'
-  '-DDHCP6_CLIENT=ON'
-  '-DOT_DUA=ON'
-  '-DOT_MLR=ON'
-  '-DOT_CSL_RECEIVER=ON'
-  '-DOT_LINK_METRICS=ON'
-  '-DBORDER_AGENT=OFF'
-  '-DOT_COAP=OFF'
-  '-DOT_COAPS=OFF'
-  '-DOT_ECDSA=OFF'
-  '-DOT_FULL_LOGS=OFF'
-  '-DOT_IP6_FRAGM=OFF'
-  '-DOT_LINK_RAW=OFF'
-  '-DOT_MTD_NETDIAG=OFF'
-  '-DOT_SNTP_CLIENT=OFF'
-  '-DOT_UDP_FORWARD=OFF')
-
-readonly BUILD_1_1_ENV=(
-  'BORDER_ROUTER=1'
-  'REFERENCE_DEVICE=1'
-  'COMMISSIONER=1'
-  'DHCP6_CLIENT=1'
-  'DHCP6_SERVER=1'
-  'JOINER=1'
-  'MAC_FILTER=1'
-  'BOOTLOADER=1'
-  'USB=1'
+if [[ -n ${BASH_SOURCE[0]} ]]; then
+    script_path="${BASH_SOURCE[0]}"
+else
+    script_path="$0"
+fi
+
+script_dir="$(dirname "$(realpath "$script_path")")"
+repo_dir="$(dirname "$script_dir")"
+
+# Global Vars
+platform=""
+build_dir=""
+
+readonly OT_PLATFORMS=(nrf52840 efr32mg1 efr32mg12 efr32mg13 efr32mg21)
+
+readonly build_1_3_options_common=(
+    ""
 )
 
-NRFUTIL=/tmp/nrfutil-linux
-if [ ! -f $NRFUTIL ]; then
-  wget -O $NRFUTIL https://github.com/NordicSemiconductor/pc-nrfutil/releases/download/v6.1/nrfutil-linux
-  chmod +x $NRFUTIL
-fi
+readonly build_1_3_options_efr32=(
+    "-DOT_SRP_SERVER=ON"
+    "-DOT_ECDSA=ON"
+    "-DOT_SERVICE=ON"
+    "-DOT_DNSSD_SERVER=ON"
+    "-DOT_SRP_CLIENT=ON"
+)
 
-if [ ! -f /tmp/private.pem ]; then
-  $NRFUTIL keys generate /tmp/private.pem
-fi
-mkdir -p "$OUTPUT_ROOT"
-
-# $1: The basename of the file to zip, e.g. ot-cli-ftd
-# $2: Thread version number, e.g. 1.2
-# $3: The binary path, defaulted as ./build-"$2"/bin/"$1"
-make_zip() {
-  local BINARY_PATH=${3:-"./build-$2/bin/$1"}
-  arm-none-eabi-objcopy -O ihex "$BINARY_PATH" "$1"-"$2".hex
-  $NRFUTIL pkg generate --debug-mode --hw-version 52 --sd-req 0 --application "$1"-"$2".hex --key-file /tmp/private.pem "$1"-"$2".zip
+readonly build_1_3_options_nrf=(
```

Comment:
Why aren't there any Thread 1.3 NRF options?

Reply:
They aren't building with any turned on so I left this options list empty

https://github.com/openthread/ot-reference-release/blob/c2566b0af0ec7853e76066bdf41bf6bc1630c1ee/script/make-firmware.bash#L123-L126

Reply:
I'm very curious how Nordic is producing 1.3 builds without turning on the config options we turned on in build_1_3_options_efr32

Are these enabled somewhere else by default? If so is our option list redundant?

Reply:
Sent an email to Jonathan asking about this.